src/kagglehub/gcs_upload.py

```diff
@@ -138,51 +147,110 @@ def _upload_blob(file_path: str, model_type: str) -> str:
+                        zipf.write(file_path, os.path.relpath(file_path, folder))
+
+            tokens = [
+                token
+                for token in [_upload_file_or_folder(temp_dir, TEMP_ARCHIVE_FILE, model_type, quiet)]
+                if token is not None
+            ]
+            return UploadDirectoryInfo(name="archive", files=tokens)
+
+    root_dict = UploadDirectoryInfo(name="root")
+    if os.path.isfile(folder):
+        # Directly upload the file if the path is a file
+        file_name = os.path.basename(folder)
+        token = _upload_file_or_folder(os.path.dirname(folder), file_name, model_type, quiet)
+        if token:
+            root_dict.files.append(token)
+    else:
+        for root, _, files in os.walk(folder):
+            # Path of the current folder relative to the base folder
+            path = os.path.relpath(root, folder)
+
+            # Navigate or create the dictionary path to the current folder
+            current_dict = root_dict
+            if path != ".":
+                for part in path.split(os.sep):
+                    # Find or create the subdirectory in the current dictionary
+                    for subdir in current_dict.directories:
+                        if subdir.name == part:
+                            current_dict = subdir
+                            break
+                    else:
+                        # If the directory is not found, create a new one
+                        new_dir = UploadDirectoryInfo(name=part)
+                        current_dict.directories.append(new_dir)
+                        current_dict = new_dir
+
+            # Add file tokens to the current directory in the dictionary
+            for file in files:
+                token = _upload_file_or_folder(root, file, model_type, quiet)
+                if token:
+                    current_dict.files.append(token)
+
+    return root_dict
+
+
+def _upload_file_or_folder(
+    parent_path: str,
+    file_or_folder_name: str,
+    model_type: str,
+    quiet: bool = False,  # noqa: FBT002, FBT001
+) -> Optional[str]:
+    """
+    Uploads a file or each file inside a folder individually from a specified path to a remote service.
+    Parameters
+    ==========
+    parent_path: The parent directory path from where the file or folder is to be uploaded.
+    file_or_folder_name: The name of the file or folder to be uploaded.
+    dir_mode: The mode to handle directories. Accepts 'zip', 'tar', or other values for skipping.
+    model_type: Type of the model that is being uploaded.
+    quiet: suppress verbose output (default is False)
+    :return: A token if the upload is successful, or None if the file is skipped or the upload fails.
+    """
+    full_path = os.path.join(parent_path, file_or_folder_name)
+    if os.path.isfile(full_path):
+        return _upload_file(file_or_folder_name, full_path, quiet, model_type)
+    elif not quiet:
+        logger.info("Skipping: " + file_or_folder_name)
```

Comment:
This will print "Skipping: <NAME of folder>" for every folder. That could be misleading for the user. They would think their folder is skipped. Should we simply not print anything?

Reply:
yes, i think that's fine